src/test/java/com/st/utopia/cancellationservice/service/TicketServiceTest.java

```diff
@@ -0,0 +1,33 @@
+package com.st.utopia.cancellationservice.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import com.st.utopia.cancellationservice.model.Ticket;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TicketServiceTest {
+	
+	@Autowired
+	private TicketService ticketservice;
+	
+    @Test
+    public void cancelReservationTest() {

```

Comment:
It looks to me like this test would pass even if `TicketService::cancelReservation` did absolutely nothing, since unless you're running against a live database that you manually initialize before each run, `ticketservice.cancelReservation` will log an error but "succeed" if the ticket was not already booked or did not exist, then `ticketservice.findTicket` will produce an essentially-empty `Ticket` object. This illustrates that methods should signal failure to their callers, not just log failure.